### PR TITLE
fix(unlock-app, paywall): a few more paywall fixes/improvements

### PR DESCRIPTION
**Pull request by julien51**
# Description

All in the title.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/paywall/src/Paywall.ts [hunk 1]
```diff
@@ -87,22 +87,25 @@ export class Paywall {
...
     if (this.iframe) {
       this.showIframe()
     } else {
-      this.shakeHands()
+      this.shakeHands(unlockUrl || unlockAppUrl)
     }
-    this.sendOrBuffer('setConfig', config || this.paywallConfig)
+    this.sendOrBuffer(
+      'setConfig',
+      injectProviderInfo(config || this.paywallConfig, this.provider)
```
Comment (julien51) [inline packages/paywall/src/Paywall.ts:107]: this was missing!

File: packages/paywall/src/Paywall.ts [hunk 2]
```diff
@@ -112,7 +115,10 @@ export class Paywall {
   resetConfig = (config: PaywallConfig) => {
     this.paywallConfig = injectProviderInfo(config, this.provider)
     this.checkKeysAndLock()
-    this.sendOrBuffer('setConfig', config || this.paywallConfig)
+    this.sendOrBuffer(
+      'setConfig',
+      injectProviderInfo(config || this.paywallConfig, this.provider)
```
Comment (julien51) [inline packages/paywall/src/Paywall.ts:120]: same here!

File: packages/paywall/src/urls.ts [hunk 3]
```diff
@@ -29,21 +29,16 @@ const endpoint = new URL(baseUrl)
 
 export function getConfigUrl(url: string) {
   let unlockAppUrl: string
-  let locksmithUri: string
```
Comment (julien51) [inline packages/paywall/src/urls.ts:32]: Not used...

File: unlock-app/src/components/interface/checkout/Connected.tsx [hunk 4]
```diff
@@ -209,6 +209,12 @@ export function Connected({
     }
   }, [autoconnect, authenticateWithProvider])
 
+  useEffect(() => {
```
Comment (julien51) [inline unlock-app/src/components/interface/checkout/Connected.tsx:212]: This makes debugging injected providers MUCH easier...

File: packages/paywall/src/Paywall.ts [hunk 5]
```diff
@@ -156,8 +162,8 @@ export class Paywall {
     return this.lockPage()
   }
 
-  shakeHands = async () => {
-    console.log(`Connecting to now ${unlockAppUrl}`)
+  shakeHands = async (unlockAppUrl: string) => {
+    console.log(`Connecting to ${unlockAppUrl}`)
```
Comment (searchableguy) [inline packages/paywall/src/Paywall.ts:166]: maybe change this to console.debug to differentiate.

File: unlock-app/src/components/interface/checkout/Connected.tsx [hunk 6]
```diff
@@ -209,6 +209,12 @@ export function Connected({
...
+    if (!account) {
+      console.log('Not connected')
```
Comment (searchableguy) [inline unlock-app/src/components/interface/checkout/Connected.tsx:214]: same here. use console.debug for debugging statements.

**Review by searchableguy — APPROVED** [review]

**Pull request merged**